weld/codegen/llvm2/jit.rs

```diff
@@ -219,12 +235,20 @@ unsafe fn verify_module(module: LLVMModuleRef) -> WeldResult<()> {
     let result_code = LLVMVerifyModule(module,
                                        LLVMReturnStatusAction,
                                        &mut error_str);
-    if result_code != 0 {
-        let err = CStr::from_ptr(error_str).to_str().unwrap();
-        compile_err!("{}", format!("Module verification failed: {}", err))
-    } else {
-        Ok(())
+    let result = {
+        if result_code != 0 {
+            let err = CStr::from_ptr(error_str)
+                .to_string_lossy()
+                .into_owned();
+            compile_err!("{}", format!("Module verification failed: {}", err))
+        } else {
+            Ok(())
+        }
+    };
+    if !error_str.is_null() {
+        libc::free(error_str as *mut libc::c_void);
```

Comment:
Move this above to before the error is returned (null check isn't needed since libc's free doesn't do anything if the pointer is null -- my mistake re: the comment earlier)

Reply:
Ah, you're right! No need to change this then (maybe get rid of the null check but not necessary)